extra-network/clash/autobuild/build

```diff
@@ -0,0 +1,5 @@
+abinfo "Building clash..."
+go build -trimpath -ldflags "-X github.com/Dreamacro/clash/constant.Version=$PKGVER" -mod=readonly
```

Comment:
How did `ABTYPE=gomod` fail?

Reply:
The `-trimpath` argument is required here, otherwise the `--version` argument of the compiled clash binary cannot display the version number correctly

Reply:
Got it, Autobuild3 does not yet handle these scenarios well.